src/adapter/src/coord/command_handler.rs

```diff
@@ -94,8 +94,8 @@ impl Coordinator {
                 outer_ctx_extra,
             } => {
                 let tx = ClientTransmitter::new(tx, self.internal_cmd_tx.clone());
-                let span = span
-                    .in_scope(|| tracing::debug_span!("message_command (execute)").or_current());
+                let span = span.or_current();
+                tracing::Span::current().add_link(span.context().span().span_context().clone());
```

Comment:
note: We could probably make this Otel linking a bit cleaner by making a `TracingSpanExt` trait, but for now I left this as-is.